frontend/src/components/Header/Header.tsx

```diff
@@ -154,6 +160,65 @@ export const Header: React.FC<Props> = ({ fullyIntegrated }) => {
 
 	const inProjectOrWorkspace = isLoggedIn && (projectId || workspaceId)
 
+	const { data: workspacesData, loading: workspacesLoading } =
+		useGetWorkspacesQuery()
+
+	const workspaceOptions = (workspacesData?.workspaces || [])
```

Comment:
Wonder if we can do this in a useMemo, to avoid calculating this on ever rerender?

Something like
```
const workspaceOptions = useMemo(() => {
  const currentWorkspaces = workspacesData?.workspaces || []
  const joinableWorkspaces = workspacesData?.joinable_workspaces || []

  return currentWorkspaces.concat(joinableWorkspaces).map((workspace) => ({
			value: workspace.id,
			displayValue: workspace.name,
			id: workspace.id,
		}))
}, [workspacesData?.workspaces, workspacesData?.joinable_workspaces])

// build JSX outside of useMemo
```